p24/src/refs.rs

```diff
@@ -0,0 +1,161 @@
+//f1: accepts mutable reference to tuple with two u32s and bool flag. If flag is false, returns mutable reference to the first element in the tuple. If flag is true, returns mutable reference to the second element in the tuple.
+pub fn f1(tuple: &mut (u32, u32, bool)) -> &mut u32 {
+    if tuple.2 {
+        &mut tuple.1
+    } else {
+        &mut tuple.0
+    }
+}
+
+//– f2: accepts mutable slice &mut [u32] and number n, returns mutable reference to the n-th element in the slice (counting from zero).
+pub fn f2(slice: &mut [u32], n: usize) -> &mut u32 {
+    &mut slice[n]
+}
+
+//– f3: accepts slice &mut [u32] and number n, returns mutable reference to the n-th element from the end of the slice (counting from zero, i.e. 0 means the last element).
+pub fn f3(slice: &mut [u32], n: usize) -> &mut u32 {
+    let len = slice.len();
+    &mut slice[len - 1 - n]
+}
+
+//– f4: accepts slice &[u32], partitions it into 4 equal (as much as possible) parts, and returns 4 resulting slices
+pub fn f4(slice: &[u32]) -> [&[u32]; 4] {
```

Comment:
Note that for slice with nine elements this function will return slices with length 3, 3, 3, and 0, which is not a fair division.